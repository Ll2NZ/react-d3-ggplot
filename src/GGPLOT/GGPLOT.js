import React              from "react";
import { Component }      from "react";
import PropTypes          from "prop-types";
import { ScalesProvider } from "../Context/";
import { WithScales }     from "../WithScales/";
import { XAxis }          from "../XAxis/";
import { YAxis }          from "../YAxis/";
import { XGrid }          from "../XGrid/";
import { YGrid }          from "../YGrid/";

class GGPLOT extends Component{
    static propTypes = {
        data: PropTypes.array,
        aes: PropTypes.array,
        scaleTypes: PropTypes.array,
        dimensions: PropTypes.shape({
            width: PropTypes.number,
            height: PropTypes.number,
            padding: PropTypes.number
        }),
        className: PropTypes.string,
        children: PropTypes.any
    }

    render(){
        const { dimensions, className } = this.props;

        return(
            <svg width={ dimensions.width } height={ dimensions.height } className={ className }>
                {/* used as background color */}
                <rect
                    width={ dimensions.width - dimensions.padding*2 }
                    height={ dimensions.height - dimensions.padding*2 }
                    fill="rgb(232, 232, 232)"
                    transform={ `translate(${dimensions.padding}, ${dimensions.padding})` }
                />

                {/* main graph */}
                <ScalesProvider value={ this.props }>
                    <WithScales>
                        {({ createScaleType }) => (
                        <>
                            <XAxis createScaleType={ createScaleType }/>
                            <XGrid createScaleType= { createScaleType }/>
                            <YAxis createScaleType={ createScaleType }/>
                            <YGrid createScaleType={ createScaleType }/>

                            {/*
                              * Here, we clone each child passed into <GGPLOT/> to
                              * get acces to createScaleType so that <GGPLOT/> is modular
                            */}
                            {React.Children.map(this.props.children, child => {
                                return React.cloneElement(child, {
                                    createScaleType: createScaleType
                                });
                            })}
                        </>
                        )}
                    </WithScales>
                </ScalesProvider>
            </svg>
        );
    }
}

export default GGPLOT;
