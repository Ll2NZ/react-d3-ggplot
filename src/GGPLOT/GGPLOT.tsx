import React, { ReactChild } from "react";
import { ChartContext } from "../_context/";
import { IContext } from "../_context";
import { Background } from "../Background/";
import { XAxis } from "../XAxis/";
import { YAxis } from "../YAxis/";
import { XGrid } from "../XGrid/";
import { YGrid } from "../YGrid/";
import includes from "lodash.includes";

interface IProps extends IContext {
  children?: ReactChild;
}

function GGPLOT(props: IProps) {
  if (!props.data || !props.aes || !props.dimensions) {
    throw new Error("GGPLOT must be supplied with the following props: data, aes, and dimensions");
  }

  if (props.children) {
    const componentNames = React.Children.map(props.children, child => {
      // @ts-ignore
      if (child.type.displayName) {
        // @ts-ignore
        return child.type.displayName; // this property exists. don't know why ts is complaining
      } else {
        return "invalid";
      }
    });

    if (includes(componentNames, "invalid")) {
      throw new Error("GGPLOT only accepts components from react-d3-ggplot");
    }
  }

  return (
    <ChartContext.Provider value={props}>
      <svg width={props.dimensions.width} height={props.dimensions.height}>
        <Background />
        <XGrid />
        <YGrid />
        <XAxis />
        <YAxis />
        {props.children}
      </svg>
    </ChartContext.Provider>
  );
}

export default GGPLOT;
