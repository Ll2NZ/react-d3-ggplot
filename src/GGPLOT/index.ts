export { default as GGPLOT } from "./GGPLOT";
