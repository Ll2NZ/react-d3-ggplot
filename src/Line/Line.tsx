import React from "react";
import { useContext } from "react";
import { ChartContext } from "../_context/";
import { useScale } from "../_hooks/";
import { line } from "d3-shape";
import { curveCatmullRom } from "d3-shape";

interface IProps {
  fill?: string;
  width?: string;
}

Line.displayName = "Line";

function Line(props: IProps) {
  const context = useContext(ChartContext);
  const xScale: any = useScale("XAxis");
  const yScale: any = useScale("YAxis");

  const lineToAppend = line()
    .x((d: any) => xScale(d[context.aes[0]]))
    .y((d: any) => yScale(d[context.aes[1]]))
    .curve(curveCatmullRom)(context.data);

  return (
    <g>
      <path
        fill="none"
        stroke={!props.fill ? "crimson" : props.fill}
        strokeWidth={!props.width ? "1" : props.width}
        d={!lineToAppend ? undefined : lineToAppend}
      />
    </g>
  );
}

export default Line;
