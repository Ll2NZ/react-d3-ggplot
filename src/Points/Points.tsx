import React from "react";
import { useContext } from "react";
import { ChartContext } from "../_context/";
import { useXScale } from "../_hooks/";
import { useYScale } from "../_hooks/";

interface IProps {
  r?: number;
  fill?: string;
}

Points.displayName = "Points";

function Points(props: IProps) {
  const context = useContext(ChartContext);
  const xScale: any = useXScale();
  const yScale: any = useYScale();

  // will remove later when testing componet
  if (!xScale || !yScale) {
    return <h1>whoops</h1>;
  }

  return (
    <g>
      {/*
        For the key, we're not supposed to used index because we can't take
        advantage of react's optimization capabilities when it comes to re-rendering.
        Will use key until we can sort this detail out.
      */}
      {context.data.map((datum, index) => (
        <circle
          key={index}
          cx={xScale(datum[context.aes[0]])}
          cy={yScale(datum[context.aes[1]])}
          r={!props.r ? 3 : props.r}
          fill={!props.fill ? "orange" : props.fill}
        />
      ))}
    </g>
  );
}

export default Points;
