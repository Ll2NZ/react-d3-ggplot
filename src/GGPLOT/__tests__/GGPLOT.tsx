import * as React from "react";
import GGPLOT from "../GGPLOT";
import { Line } from "../../Line/";
import { Points } from "../../Points/";
// need to test rects first before we can us it!
// import { Rects } from "../../Rects/";
import { render, cleanup } from "react-testing-library";

afterEach(cleanup);

const validState = {
  data: [{ x: 1, y: 1 }, { x: 100, y: 100 }],
  aes: ["x", "y"],
  dimensions: { width: 300, height: 500, padding: 10 }
};

const invalidState = {
  // data: [{ x: 1, y: 1 }, {x: 100, y: 100}],
  aes: ["x", "y"],
  dimensions: { width: 300, height: 500, padding: 10 }
};

test("Passing an invalid state shape should fail", () => {
  // @ts-ignore
  const node = () => render(<GGPLOT {...invalidState} />);

  expect(node).toThrow();
});

test("Passing no child into GGPLOT should work just fine and render only the XAxis and YAxis", () => {
  const { container } = render(<GGPLOT {...validState} />);

  // the underlying SVG markup generated above should not change when we change our code base logic!
  expect(container.firstChild).toMatchSnapshot();
});

test("GGPLOT can only accept one of the following: Line, Points, and Rects", () => {
  const firstNode = () =>
    render(
      <GGPLOT {...validState}>
        <div>Invalid child</div>
      </GGPLOT>
    );
  expect(firstNode).toThrow();

  const secondNode = () =>
    render(
      // @ts-ignore
      <GGPLOT {...validState}>
        <div />
        <div />
        <div />
      </GGPLOT>
    );
  expect(secondNode).toThrow();
});

test("GGPLOT accepting ONE valid component", () => {
  const firstNode = render(
    <GGPLOT {...validState}>
      <Line />
    </GGPLOT>
  );
  // the underlying SVG markup generated above should not change when we change our code base logic!
  expect(firstNode.container.firstChild).toMatchSnapshot();

  const secondNode = render(
    <GGPLOT {...validState}>
      <Points />
    </GGPLOT>
  );
  // the underlying SVG markup generated above should not change when we change our code base logic!
  expect(secondNode.container.firstChild).toMatchSnapshot();
});
