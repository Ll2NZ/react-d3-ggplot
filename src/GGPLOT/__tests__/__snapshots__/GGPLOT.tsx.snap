// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GGPLOT accepting ONE valid component 1`] = `
<svg
  height="500"
  width="300"
>
  <rect
    fill="#f1f1f1"
    height="480"
    transform="translate(10, 10)"
    width="280"
  />
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="middle"
    transform="translate(0, 10)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(10.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(66.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(122.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(178.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(234.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(290.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="start"
    transform="translate(10, 0)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(0,490.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,394.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,298.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,202.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,106.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,10.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="middle"
    transform="translate(0, 490)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(10.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        0
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(38.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        10
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(66.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        20
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(94.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        30
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(122.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        40
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(150.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        50
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(178.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        60
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(206.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        70
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(234.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        80
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(262.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        90
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(290.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        100
      </text>
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="end"
    transform="translate(10, 0)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(0,490.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        0
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,442.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        10
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,394.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        20
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,346.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        30
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,298.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        40
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,250.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        50
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,202.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        60
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,154.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        70
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,106.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        80
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,58.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        90
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,10.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        100
      </text>
    </g>
  </g>
  <g>
    <path
      d="M12.8,485.2L290,10"
      fill="none"
      stroke="crimson"
      stroke-width="1"
    />
  </g>
</svg>
`;

exports[`GGPLOT accepting ONE valid component 2`] = `
<svg
  height="500"
  width="300"
>
  <rect
    fill="#f1f1f1"
    height="480"
    transform="translate(10, 10)"
    width="280"
  />
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="middle"
    transform="translate(0, 10)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(10.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(66.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(122.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(178.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(234.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(290.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="start"
    transform="translate(10, 0)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(0,490.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,394.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,298.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,202.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,106.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,10.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="middle"
    transform="translate(0, 490)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(10.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        0
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(38.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        10
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(66.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        20
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(94.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        30
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(122.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        40
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(150.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        50
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(178.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        60
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(206.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        70
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(234.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        80
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(262.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        90
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(290.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        100
      </text>
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="end"
    transform="translate(10, 0)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(0,490.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        0
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,442.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        10
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,394.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        20
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,346.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        30
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,298.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        40
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,250.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        50
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,202.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        60
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,154.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        70
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,106.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        80
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,58.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        90
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,10.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        100
      </text>
    </g>
  </g>
  <g>
    <circle
      cx="12.8"
      cy="485.2"
      fill="orange"
      r="3"
    />
    <circle
      cx="290"
      cy="10"
      fill="orange"
      r="3"
    />
  </g>
</svg>
`;

exports[`Passing no child into GGPLOT should work just fine and render only the XAxis, YAxis, XGrid, YGrid, and Background 1`] = `
<svg
  height="500"
  width="300"
>
  <rect
    fill="#f1f1f1"
    height="480"
    transform="translate(10, 10)"
    width="280"
  />
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="middle"
    transform="translate(0, 10)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(10.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(66.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(122.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(178.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(234.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(290.5,0)"
    >
      <line
        opacity="0.7"
        stroke="white"
        y2="480"
      />
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="start"
    transform="translate(10, 0)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(0,490.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,394.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,298.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,202.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,106.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,10.5)"
    >
      <line
        opacity="0.7"
        stroke="white"
        x2="280"
      />
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="middle"
    transform="translate(0, 490)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(10.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        0
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(38.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        10
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(66.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        20
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(94.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        30
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(122.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        40
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(150.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        50
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(178.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        60
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(206.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        70
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(234.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        80
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(262.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        90
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(290.5,0)"
    >
      <line
        stroke="currentColor"
        y2="6"
      />
      <text
        dy="0.71em"
        fill="currentColor"
        y="9"
      >
        100
      </text>
    </g>
  </g>
  <g
    fill="none"
    font-family="sans-serif"
    font-size="10"
    text-anchor="end"
    transform="translate(10, 0)"
  >
    <g
      class="tick"
      opacity="1"
      transform="translate(0,490.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        0
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,442.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        10
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,394.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        20
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,346.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        30
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,298.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        40
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,250.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        50
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,202.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        60
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,154.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        70
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,106.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        80
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,58.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        90
      </text>
    </g>
    <g
      class="tick"
      opacity="1"
      transform="translate(0,10.5)"
    >
      <line
        stroke="currentColor"
        x2="-6"
      />
      <text
        dy="0.32em"
        fill="currentColor"
        x="-9"
      >
        100
      </text>
    </g>
  </g>
</svg>
`;
